js/hand-poser.js

```diff
@@ -16,69 +18,95 @@ function mirror(q) {
-        "ring-finger-metacarpal", "ring-finger-phalanx-proximal", "ring-finger-phalanx-intermediate", "ring-finger-phalanx-distal", "ring-finger-tip",
-        "pinky-finger-metacarpal", "pinky-finger-phalanx-proximal", "pinky-finger-phalanx-intermediate", "pinky-finger-phalanx-distal", "pinky-finger-tip"
-    ],
-
-    start: function() {
+export class HandPose extends Component {
+    static TypeName = 'hand-pose';
+    static Properties = {
+        jsonFile: {type: Type.String, default: ''},
+    };
+
+    ORDERED_JOINTS = [
+        'wrist',
+        'thumb-metacarpal',
+        'thumb-phalanx-proximal',
+        'thumb-phalanx-distal',
+        'thumb-tip',
+        'index-finger-metacarpal',
+        'index-finger-phalanx-proximal',
+        'index-finger-phalanx-intermediate',
+        'index-finger-phalanx-distal',
+        'index-finger-tip',
+        'middle-finger-metacarpal',
+        'middle-finger-phalanx-proximal',
+        'middle-finger-phalanx-intermediate',
+        'middle-finger-phalanx-distal',
+        'middle-finger-tip',
+        'ring-finger-metacarpal',
+        'ring-finger-phalanx-proximal',
+        'ring-finger-phalanx-intermediate',
+        'ring-finger-phalanx-distal',
+        'ring-finger-tip',
+        'pinky-finger-metacarpal',
+        'pinky-finger-phalanx-proximal',
+        'pinky-finger-phalanx-intermediate',
+        'pinky-finger-phalanx-distal',
+        'pinky-finger-tip',
+    ];
+
+    start() {
         this.hand = this.object.getComponent('hand-tracking');
-        if(this.jsonFile != "")
-            fetch(this.jsonFile).then(function(d) { d.json(); }).then(this.onHandPoseDownloaded.bind(this));
-    },
-
-    mirrorPose: function(out, pose) {
-        for(let key in pose) out[key] = mirror(pose[key]);
+        if (this.jsonFile != '')
+            fetch(this.jsonFile)
+                .then(function (d) {
+                    d.json();
+                })
+                .then(this.onHandPoseDownloaded.bind(this));
+    }
+
+    mirrorPose(out, pose) {
+        for (let key in pose) out[key] = mirror(pose[key]);
         return out;
-    },
+    }
 
     /* Create and download a snapshot of the current hand pose */
-    snapshot: function() {
+    snapshot() {
         const children = this.object.children;
 
         const snapshot = {};
-        for(let i = 0; i < children.length; ++i) {
+        for (let i = 0; i < children.length; ++i) {
             const o = children[i];
             const name = o.name;
             /* We only snapshot bone poses, not the mesh or other attachments */
-            if(!this.ORDERED_JOINTS.includes(name)) continue;
-            if(name.endsWith('wrist')) continue;
+            if (!this.ORDERED_JOINTS.includes(name)) continue;
+            if (name.endsWith('wrist')) continue;
 
             /* Have the snapshot independent of handedness, save right hand only */
             snapshot[name.substr(4)] = Array.from(o.transformLocal);
```

Comment:
```suggestion
            snapshot[name.substr(4)] = o.getTransformLocal(new Float32Array(8));
```

Since `.transformLocal` is now deprecated 😌